cdk/resources/WifiSiteSurveyGeolocation.ts

```diff
@@ -0,0 +1,173 @@
+				resultPath: '$.wifisurveygeo',
+				payloadResponseOnly: true,
+			},
+		)
+
+		const persistResult = new StepFunctionTasks.DynamoUpdateItem(
+			this,
+			'Update location from nRF cloud',
+			{
+				table: storage.surveysTable,
+				key: {
+					surveyId: StepFunctionTasks.DynamoAttributeValue.fromString(
+						StepFunctions.JsonPath.stringAt('$.surveyId'),
+					),
+				},
+				updateExpression:
+					'SET #unresolved = :unresolved, #lat= :lat, #lng = :lng, #accuracy = :accuracy, #updatedAt = :updatedAt',
+				expressionAttributeNames: {
+					'#unresolved': 'unresolved',
+					'#lat': 'lat',
+					'#lng': 'lng',
+					'#accuracy': 'accuracy',
+					'#updatedAt': 'updatedAt',
+				},
+				expressionAttributeValues: {
+					':unresolved':
+						StepFunctionTasks.DynamoAttributeValue.fromBoolean(false),
+					':lat': StepFunctionTasks.DynamoAttributeValue.numberFromString(
+						// It seems it is a bug from CDK, https://github.com/aws/aws-cdk/issues/12456
+						StepFunctions.JsonPath.stringAt(
+							`States.Format('{}', $.wifisurveygeo.lat)`,
+						),
+					),
+					':lng': StepFunctionTasks.DynamoAttributeValue.numberFromString(
+						StepFunctions.JsonPath.stringAt(
+							`States.Format('{}', $.wifisurveygeo.lng)`,
+						),
+					),
+					':accuracy': StepFunctionTasks.DynamoAttributeValue.numberFromString(
+						StepFunctions.JsonPath.stringAt(
+							`States.Format('{}', $.wifisurveygeo.accuracy)`,
+						),
+					),
+					':updatedAt': StepFunctionTasks.DynamoAttributeValue.fromString(
+						StepFunctions.JsonPath.stringAt('$$.State.EnteredTime'),
+					),
+				},
+			},
+		).next(
+			new StepFunctions.Succeed(this, 'Done (resolved using nRF Cloud API)'),
+		)
+
+		const persistFailure = new StepFunctionTasks.DynamoUpdateItem(
+			this,
+			'Update resolution failure',
+			{
+				table: storage.surveysTable,
+				key: {
+					surveyId: StepFunctionTasks.DynamoAttributeValue.fromString(
+						StepFunctions.JsonPath.stringAt('$.surveyId'),
+					),
+				},
+				updateExpression:
+					'SET #unresolved = :unresolved, #updatedAt = :updatedAt',
+				expressionAttributeNames: {
+					'#unresolved': 'unresolved',
+					'#updatedAt': 'updatedAt',
+				},
+				expressionAttributeValues: {
+					':unresolved':
+						StepFunctionTasks.DynamoAttributeValue.fromBoolean(true),
+					':updatedAt': StepFunctionTasks.DynamoAttributeValue.fromString(
+						StepFunctions.JsonPath.stringAt('$$.State.EnteredTime'),
+					),
+				},
+			},
+		).next(
+			new StepFunctions.Fail(this, 'Failed (no resolution)', {
+				error: 'FAILED',
+				cause: 'The Wifi survey request could not be resolved',
+			}),
+		)
+
+		const checkAPIResult = new StepFunctions.Choice(
+			this,
+			'Resolved from nRF Cloud API?',
+		)
+			.when(
+				StepFunctions.Condition.booleanEquals(`$.wifisurveygeo.located`, true),
+				persistResult,
+			)
+			.otherwise(persistFailure)
+
+		const definition = resolveWifiSurvey.next(checkAPIResult)
```

Comment:
add a check if it has already been resolved, see https://github.com/NordicSemiconductor/asset-tracker-cloud-aws-js/blob/2e2f7497f82d6756faeffdababfefaf6bfdba91b/cdk/resources/PGPSResolver.ts#L191

Reply:
I am not sure whether we have to check it because 

1.  Step function runs once. Of course, lambda http api will execute the step function when location has not been resolved it yet
2. In case of concurrent requests, step function will handle this case
3. In edge case, the location was resolved before executing step function. We can pay only another execution and it will be protected by step function for another 90 days.

What is your opinion?

Reply:
Yes, right in this case it doesn't add additional protection